src/mdyml/convert_cisagov.py

```diff
@@ -0,0 +1,209 @@
+    0   No error occurred.
+    >0  An error occurred.
+
+Usage:
+  convert-cisagov [--log-level=LEVEL]
+  convert-cisagov (-h | --help)
+
+Options:
+  -h --help              Show this message.
+  --log-level=LEVEL      If specified, then the log level will be set to
+                         the specified value.  Valid values are "debug", "info",
+                         "warning", "error", and "critical". [default: info]
+"""
+
+# Standard Python Libraries
+from datetime import datetime, timezone
+import html
+import logging
+import sys
+from typing import Any
+import urllib.request
+
+# Third-Party Libraries
+import dateparser
+import docopt
+import ruamel.yaml
+from schema import And, Schema, SchemaError, Use
+
+from . import DEFAULT_CVE_ID, MD_LINK_RE, ORDERED_CVE_IDS, __version__
+
+RAW_URL = "https://raw.githubusercontent.com/cisagov/log4j-affected-db/develop/SOFTWARE-LIST.md"
+
+EXPECTED_COLUMN_NAMES = [
+    "vendor",
+    "product",
+    "versions",
+    "status",
+    "update_available",  # This column will be ignored as it can be inferred from patched_versions
+    "vendor_link",
+    "notes",
+    "references",
+    "last_updated",
+]
+EXPECTED_COLUMN_COUNT = len(EXPECTED_COLUMN_NAMES)
+
+
+def convert() -> None:
+    """Parse the Markdown at the given URL and convert it to YAML."""
+    # Get the Markdown
+    logging.info("Reading Markdown at %s", RAW_URL)
+    # We are using a hardcoded URL so there is no danger of unexpected schemes.
+    response = urllib.request.urlopen(RAW_URL)  # nosec
+
+    # Read rows from all tables
+    table_rows = []
+    in_table = False
+    skip_next_line = False
+    for raw_line in response:
+        line = html.unescape(raw_line.decode("utf-8"))
+        logging.debug("Read line: %s", line)
+        if skip_next_line:
+            skip_next_line = False
+            continue
+        if not in_table and line.startswith("|"):
+            logging.debug("Table start detected")
+            # Remove any trailing newline, any beginning or ending pipes, and
+            # split the remaining value on pipes.
+            possible_header_columns = line.rstrip().strip("|").split("|")
+            if len(possible_header_columns) == EXPECTED_COLUMN_COUNT:
+                in_table = True
+                skip_next_line = True
+            continue
+        if in_table and not line.startswith("|"):
+            logging.warning("Line does not start with |, skipping: %s", line)
+            continue
+        if in_table:
+            logging.debug("Appending row")
+            table_rows.append(line)
+    logging.info("Done reading markdown table %d rows.", len(table_rows))
+
+    # Process all the data into a list of dictionaries
+    out_dict_list = []
+    row_count = 0
+    for row in table_rows:
+        row_count += 1
+        in_row_list = row.split("|")[1:-1]
+        logging.debug(
+            "Processing line %d with %d columns: %s", row_count, len(in_row_list), row
```

Comment:
This number does not correspond to a line in the source file.

```suggestion
            "Processing row %d with %d columns: %s", row_count, len(in_row_list), row
```